ios/Classes/FlutterPlayer.swift

```diff
@@ -152,10 +163,20 @@ private extension FlutterPlayer {
         }
 
         let source: Source
-        if let sourceMetadata {
-            source = SourceFactory.create(from: sourceConfig, sourceMetadata: sourceMetadata)
+        if let sourceMetadata = flutterSource.sourceConfig.analyticsSourceMetadata {
+            source = SourceFactory.create(
+                from: flutterSource.sourceConfig.config,
+                sourceMetadata: sourceMetadata
+            )
         } else {
-            source = SourceFactory.create(from: sourceConfig)
+            source = SourceFactory.create(from: flutterSource.sourceConfig.config)
+        }
+
+        if let castSourceConfig = flutterSource.remoteControl?.castSourceConfig {
+            player.config.remoteControlConfig.prepareSource = { type, sourceConfig in
+                guard type == .cast else { return sourceConfig }
+                return castSourceConfig.config
+            }
         }
```

Comment:
~Is the playlist feature supported? 🤔~ This approach will always override the Source.

Reply:
Good catch: 7540e42d